### PR TITLE
Import reduce shared

### PR DESCRIPTION
**Pull request by maxtremblay**
- Import the shared reduction from Burn.
- Add support for Line operation.
- Implement a bunch of tests (also update the test framework to be robust against the small precision of bf16 and f16)
- Impose that ArgMax and ArgMin always returns the smallest coordinate in case of item equality.

### CODE REVIEW

File: crates/cubecl-reduce/src/shared.rs [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+            let destination = jump * 2 * UNIT_POS;
+            let origin = jump * (2 * UNIT_POS + 1);
+            if UNIT_POS < num_active_units {
+                RD::merge(&mut accumulator, destination, origin);
+            }
+            jump *= 2;
+            sync_units();
+        }
+    } else {
+        // 25
```
Comment (louisfd) [inline crates/cubecl-reduce/src/shared.rs:114]: Maybe remove this 25 and the 2* UP

File: crates/cubecl-reduce/src/shared.rs [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+        }
+    }
+
+    if UNIT_POS == 0 {
+        RD::write_first(output, &accumulator, CUBE_POS, input.shape(reduce_dim));
+    }
+}
+
+#[cube]
+fn div_ceil(a: u32, b: u32) -> u32 {
```
Comment (louisfd) [inline crates/cubecl-reduce/src/shared.rs:135]: std material

**Review by louisfd — APPROVED** [review]
LGTM, just some comments to cleanup

File: crates/cubecl-reduce/src/shared.rs (hunk 1 above)
Comment (maxtremblay) [reply crates/cubecl-reduce/src/shared.rs:114]: Ha yes. They were notes for me to follow the algorithm :)

**Pull request merged**